addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/discovery/SamsungTvDiscoveryParticipant.java

```diff
@@ -78,29 +91,26 @@ private String getLabel(RemoteDevice device) {
 
     @Override
     public ThingUID getThingUID(RemoteDevice device) {
-        if (device != null) {
+        String manufacturer = device.getDetails().getManufacturerDetails().getManufacturer();
+        String modelName = device.getDetails().getModelDetails().getModelName();
+        String friendlyName = device.getDetails().getFriendlyName();
```

Comment:
Do you know if any of all these chained methods can return null? If so it might result in an NPE.

Reply:
I don't know for certain. It doesn't happen on my system and this code was not written by me. I will add some extra checks.